### PR TITLE
Adding tests for splitPoints and splitEdges in Parser, cleaning up test extensions

### PR DESCRIPTION
**Pull request by cixzhang**
Includes a few fixes to grabbing the half edges from names in parser, and fixing splitEdges to be more correct.

I've updated geometry.equivalent to check arrays of any size as well as single numbers.

For testing splitPoints and splitEdges, I had to modify the assertion library to allow testing for "equivalent/closeTo" vectors in function arguments in the spy. I also moved our "closeTo" override into the chai extension as "equivalentTo" ...and also it uses geometry.equivalent. >_>

### CODE REVIEW

File: test/parser.js [hunk 1]
```diff
@@ -49,8 +57,54 @@ describe('parser', function () {
...
+    checkPaper(paper);
+    var basicFold = chai.spy.on(paper, 'basicFold');
+    parser.perform('fold ' + eA + ' over to ' + eB);
+    expect(basicFold).to.have.been.called.withSimilar([0.25, 0.75, 0]);
+    expect(basicFold).to.have.been.called.withSimilar([Math.sqrt(2)/2, -Math.sqrt(2)/2, 0]);
     expect(basicFold).to.have.been.called.with(true);
+    checkPaper(paper);
   });
+
+  function checkPaper (paper) {
```
Comment (cixzhang) [inline test/parser.js]: Added this function just to see some of the half edge structure before and after the folds. Take a look and see if it looks right to you. I can remove this afterwards :D

Comment (ptpham) [inline test/parser.js]: Does this look at a single face only? It doesn't look like it traverse any mirrors?

Comment (cixzhang) [inline test/parser.js]: Yep, single face.

Comment (ptpham) [inline test/parser.js]: Hmm ... I think it's a good idea to have more ability to check the paper, but this seems a bit special case to me? It prints only the initial face of a paper?

Comment (ptpham) [inline test/parser.js]: Also, this may have been more succinctly expressed as _.map(initial.face, console.log);

Comment (ptpham) [inline test/parser.js]: Er ... sorry, that's wrong.

Comment (ptpham) [inline test/parser.js]: More like ... _.chain(initial.face).pluck('src').each(logSrc).value().

Comment (cixzhang) [inline test/parser.js]: Er, this is just a temporary function that shows a bit of the half edge structure in the tests. :P I wasn't planning on keeping it..but we could.

Comment (ptpham) [inline test/parser.js]: Got it. I think the functionality is really important, but this form is a bit strange. XD

File: chai.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+
+var chai = require('chai');
+var spies = require('chai-spies');
+var EPSILON = require('./constants').EPSILON;
+
+// Yeah..geometry...
+var equivalent = require('./geometry').equivalent;
+
```
Comment (ptpham) [inline chai.js]: YES! Good times. I wanted to do something more native to Chai, but I haven't had the time to do it. Also, I'd like to encourage you to refrain from comments with significantly emotion content and/or low information content. =P =P =P =P

Comment (ptpham) [inline chai.js]: /// significant emotional content.

File: chai.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          if (equivalent(callArg, arg)) found++;
+        }
+      });
+      if (found === args.length) passed++;
+    });
+
+    this.assert(
+        (always && passed === calls.length) || passed > 0
+      , 'expected ' + this._obj + ' to have been called with something similar to #{exp} but got ' + passed + ' out of ' + calls.length
+      , 'expected ' + this._his + ' to have not been called with something similar to #{exp} but got ' + passed + ' times'
```
Comment (ptpham) [inline chai.js]: I'm assuming these underscore properties part of the public API?

File: geometry.js [hunk 4]
```diff
@@ -74,10 +77,17 @@ function direction (a, b) {
...
-  return Math.abs(a[0] - b[0]) <= EPSILON &&
-      Math.abs(a[1] - b[1]) <= EPSILON &&
-      Math.abs(a[2] - b[2]) <= EPSILON;
+function equivalent (a, b, delta) {
+  delta = delta || EPSILON;
+  if (a instanceof Float32Array || a instanceof Array) {
+    var flag = true;
+    for (var i = 0; i < a.length; i++) {
+      flag = flag && equivalent(a[i], b[i], delta);
+      if (!flag) break;
```
Comment (ptpham) [inline geometry.js]: Could you just do if (!equivalent(...)) return false; and then return true at the end of the type-checking block and return false at the end of the function?

Comment (ptpham) [inline geometry.js]: (principle of minimizing state).

File: chai.js (hunk 3 above)
Comment (cixzhang) [inline chai.js]: _obj is in the documentation. _his was a mystery.

**Pull request merged**